pkg/e2e/verify/namespace_webhook.go

```diff
@@ -157,6 +186,80 @@ var _ = ginkgo.Describe(namespaceWebhookTestName, func() {
 				Expect(err).NotTo(HaveOccurred())
 			}
 		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("dedicated admin can not create PrometheusRule", func(ctx context.Context) {
+			for privilegedNamespace := range PRIVILEGED_NAMESPACES {
+				prome2etestrule := e2eCreatePrometheusRule(prometheusName, privilegedNamespace, DUMMY_USER, "dedicated-admins", h)
+				err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+					_, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(ctx, prome2etestrule, metav1.CreateOptions{})
+					if err != nil && errors.IsForbidden(err) {
+						log.Printf("failed creating prometheus rules: %v", err)
+						return true, nil
+					}
+					return false, fmt.Errorf("prometheusrule could be created")
+				})
+				Expect(errors.IsForbidden(err)).To(Equal(true))
+			}
+		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("Non-privileged users cannot create prometheusrule in privileged namespaces", func(ctx context.Context) {
+			for privilegedNamespace := range PRIVILEGED_NAMESPACES {
+				prome2etestrule := e2eCreatePrometheusRule(prometheusName, privilegedNamespace, DUMMY_USER, DUMMY_GROUP, h)
+				err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+					_, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(ctx, prome2etestrule, metav1.CreateOptions{})
+					if err != nil && errors.IsForbidden(err) {
+						log.Printf("failed creating prometheus rules: %v", err)
+						return true, nil
+					}
+					return false, fmt.Errorf("prometheusrule could be created")
```

Comment:
Return the actual error here rather than creating a new one, that way if `err` is nil it will actually trigger the re-poll.

```suggestion
					return false, err
```